tools/wakucanary/wakucanary.nim

```diff
@@ -0,0 +1,127 @@
+import
+  std/strutils,
+  confutils,
+  chronos,
+  stew/shims/net
+import
+  libp2p/protocols/ping,
+  libp2p/crypto/[crypto, secp]
+import
+  ../../waku/v2/node/peer_manager/peer_manager,
+  ../../waku/v2/utils/peers,
+  ../../waku/v2/node/wakunode2,
+  ../../waku/v2/node/waku_payload,
+  ../../waku/v2/utils/peers
+
+# protocol each node shall support
+const store_prot = "/vac/waku/store/"
+const static_prot = "/vac/waku/relay/"
+const lightpush_prot = "/vac/waku/lightpush/"
+const filter_prot = "/vac/waku/filter/"
+
+# cli flags
+type
+  WakuCanaryConf* = object
+    staticnode* {.
+      desc: "Multiaddress of a static node to attemp to dial",
+      defaultValue: "",
+      name: "staticnode" }: string
+
+    storenode* {.
+      desc: "Multiaddress of a store node to attemp to dial",
+      defaultValue: "",
+      name: "storenode" }: string
```

Comment:
Why do we use two different flags for providing the address? Seems weird.

If we want to handle checking different protocols why not something like:
```sh
./wakucanary --address=${ADDR} --protocols=store,relay
```
This way we can also easily extend it in the future to check for other protocols right?

Reply:
Good point, agree.

Reply:
I like this idea! I think the way it's now was to mimic what we do in the general `wakunode2` config and the [v1 canary service](https://github.com/status-im/status-go/tree/dde0c71cfc7de711c432689ec2f38b8c96eed958/cmd/node-canary), but this feels more natural and extensible. We could potentially test multiple nodes for multiple protocols without needing complicated config.

Reply:
> We could potentially test multiple nodes for multiple protocols without needing complicated config.

You mean "in go one go"? I would limit the tool to test just one node. Since the cli flags can get really messy. I prefer to do this to test multiple nodes:

```console
./wakucanary --address=node1 --protocols=store,relay
./wakucanary --address=node2 --protocols=store,relay
./wakucanary --address=node3 --protocols=store,relay
```

Than this, which imho is a bit convoluted:
```console
./wakucanary --address=node1,node2,node3 --protocols=[[store,relay], [store,relay][store,relay]]

or 

./wakucanary --address=node1 --address=node2 --address=node3 --protocols=store,relay --protocols=store,relay --protocols=store,relay
```

Reply:
Right, I agree and would make the exit code more difficult to interpret. Same argument could probably go for testing multiple protocols simultaneously, but I guess this depends on use case. In terms of the `--protocols=` config item, perhaps you want to make this a repeated `--protocol=` argument rather than a comma separated list? Saves a bit of parsing and is already supported in nim-config to return a `seq`. See for example: https://github.com/status-im/nwaku/blob/master/waku/v2/node/config.nim#L177-L179

Reply:
I have no use for a case in which we provide multiple addresses.

Reply:
And I don't care if protocols are separate flags or a list.

Reply:
You're over-complicating it by trying to support multiple addresses.

Reply:
Agree. One address.

Reply:
And as for exit code. If I specify protocol `a` and `b` I assume I want both, if either fails the exit code should be non-zero.

The log messages should of course indicate which ones failed and which did not, but if any do fail, the whole canary check has failed and the exit code should be non-zero.

Reply:
All addressed in: c4c6da28693575b9f32aae29995a59f91ccfa083